src/Sylius/Channel/ProChannelContext.php

```diff
@@ -0,0 +1,34 @@
+<?php
+declare(strict_types=1);
+
+namespace AppBundle\Sylius\Channel;
+
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Channel\Model\ChannelInterface;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class ProChannelContext implements ChannelContextInterface
+{
+    /**
+     * @var ChannelRepositoryInterface
+     */
+    private ChannelRepositoryInterface $channelRepository;
+    /**
+     * @var Request|null
+     */
+    private ?Request $masterRequest;
+
+    public function __construct(ChannelRepositoryInterface $channelRepository, RequestStack $request)
+    {
+
+        $this->channelRepository = $channelRepository;
+        $this->masterRequest = $request->getMasterRequest();
+    }
+
+    public function getChannel(): ChannelInterface
+    {
+        return $this->channelRepository->findOneByCode($this->masterRequest->cookies->get('channel_cart', 'web'));
```

Comment:
When the channel does not exist, the following error occurs:

```
Return value of AppBundle\Sylius\Channel\ProChannelContext::getChannel() must implement 
interface Sylius\Component\Channel\Model\ChannelInterface, null returned
```

When the channel does not exist, it should throw a `ChannelNotFoundException`, for the chain to execute properly.
See [CompositeChannelContext](https://github.com/Sylius/Sylius/blob/34fcad9732bb53eab1d02a721900bb2b3e1edda2/src/Sylius/Component/Channel/Context/CompositeChannelContext.php)